demos/common/processing/config_factory.h

```diff
@@ -0,0 +1,37 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#pragma once
+#include <map>
+#include <string>
+#include "gflags/gflags.h"
+
+struct CnnConfig{
+    std::string devices;
+    std::string cpuExtensionsPath;
+    std::string clKernelsConfigPath;
+    unsigned int maxAsyncRequests=2;
```

Comment:
You already have a default value for FLAGS_nireq. Don't introduce another one here. And report an error if `FLAGS_nireq` is empty. As a later step it is possible to add an heuristic into `ConfigFactory::GetCommonConfig()` to pick good nireq value.

Reply:
I've removed default values. But FLAGS_nireq parameter still has default value of 2 (that was introduced in your code BTW).

Reply:
That's right `FLAGS_nireq` should have default value. `maxAsyncRequests` shouldn't.